jmespath-core/src/main/java/io/burt/jmespath/Adapter.java

```diff
@@ -82,6 +94,12 @@
   T getProperty(T value, String name);
 
   /**
+   * Returns all of the property names of the given object, or an empty list
+   * when the given value is not an object.
+   */
+  Collection<String> getPropertyNames(T value);
```

Comment:
If it turns out that the only use of this method is to support the `keys` function I will revisit it and replace it with something that returns `T` (with a documentation requirement that it's a JSON array of strings).